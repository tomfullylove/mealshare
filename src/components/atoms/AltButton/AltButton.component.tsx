import React from 'react';

import {Container, Text} from './assets/styles';

interface Props {
  alt: boolean,
  onPress: () => void;
  text: string;
}

const AltButton: React.FC<Props> = ({alt, onPress, text}) => {
  return (
    <Container alt={alt} onPress={onPress}>
      <Text alt={alt}>{text}</Text>
    </Container>
  );
};

export default AltButton;
