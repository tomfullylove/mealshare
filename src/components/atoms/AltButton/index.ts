import AltButton from './AltButton.component';

export default AltButton;
