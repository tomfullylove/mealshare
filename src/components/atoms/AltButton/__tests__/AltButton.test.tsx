import React from 'react';
import {shallow} from 'enzyme';

import AltButton from '../AltButton.component';

describe('AltButton', () => {
  it('renders correctly for one error', () => {
    const component = shallow(
      <AltButton onPress={jest.fn()} text="Press Me" />,
    );
    expect(component).toMatchSnapshot();
  });
});
