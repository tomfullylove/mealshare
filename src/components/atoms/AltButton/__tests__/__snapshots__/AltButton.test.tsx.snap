// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AltButton renders correctly for one error 1`] = `
<Styled(Component)
  onPress={[MockFunction]}
>
  <Styled(Text)>
    Press Me
  </Styled(Text)>
</Styled(Component)>
`;
